docs/source-2.0/tutorials/full-stack-tutorial.rst

```diff
@@ -0,0 +1,793 @@
+model in its :ref:`JSON AST representation <json-ast>`, and a ``sources`` directory which contains the model files used
+in the build. Additional artifacts can be produced by configuring plugins, and
+:doc:`code-generators <../guides/using-code-generation/index>` are prime examples of this.
+
+Generating the Server SDK
+=========================
+The server SDK is a code-generated component which provides built-in serialization, request-handling, and
+scaffolding (or "stubs") for our service as it is modeled. It facilitates the implementation of the service by
+providing these things, and allowing the implementer to focus on the business logic. Let's generate the server SDK
+for our service by adding the following build configuration:
+
+.. code-block:: json
+    :caption: ``smithy-build.json``
+
+    {
+        "version": "1.0",
+        "maven": {
+            "dependencies": [
+                "software.amazon.smithy:smithy-aws-traits:1.50.0",
+                "software.amazon.smithy:smithy-validation-model:1.50.0",
+                "software.amazon.smithy.typescript:smithy-aws-typescript-codegen:0.22.0"
+            ]
+        },
+        "plugins": {
+            "typescript-ssdk-codegen": {
+                "package" : "@com.example/coffee-service-server",
+                "packageVersion": "0.0.1"
+            }
+        }
+    }
+
+Run the build:
+
+.. code-block:: sh
+    :caption: ``/bin/sh``
+
+    smithy build model/
+
+The will should fail for the following reason:
+
+.. code-block:: text
+    :caption: ``failure message``
+
+    Projection source failed: software.amazon.smithy.codegen.core.CodegenException:
+        Every operation must have the smithy.framework#ValidationException error attached
+            unless disableDefaultValidation is set to 'true' in the plugin settings.
+        Operations without smithy.framework#ValidationException errors attached:
+            [com.example#CreateOrder, com.example#GetMenu, com.example#GetOrder]
+
+
+The server SDK validates inputs by default, and therefore enforces that each operation has the
+``smithy.framework#ValidationException`` attached to it. We can easily add this to our model by attaching the error
+to our service, meaning that all operations in the service closure may return it. Let's do this now:
+
+.. TODO: does this need a better explanation?
+
+.. code-block:: smithy
+    :caption: ``main.smithy``
+
+    use aws.protocols#restJson1
+    use smithy.framework#ValidationException
+
+    ...
+    service CoffeeShop {
+        ...
+        errors: [
+            ValidationException
+        ]
+    }
+
+
+After fixing this and running the build, the TypeScript code-generator plugin will have created a new
+artifact under ``build/smithy/source/typescript-ssdk-codegen``. This artifact contains the generated server SDK, and
+we are now able to use it in our back-end.
+
+.. TODO: Not sure if we should, but a brief look at the generated code might be good?
```

Comment:
I think it would be a good idea.

Reply:
After pondering, we decided this probably isn't a good idea - the generated code may confuse the reader because of its verbosity.